### PR TITLE
feat: add  troubleshooting page

### LINKED ISSUE #2858: Troubleshooting page
### Is your enhancement related to a problem? Please describe

Sometimes Podman Desktop does not list the containers or users have the feeling it's stuck

### Describe the solution you'd like

Display a troubleshooting page 

Users could list the 'backend state" of Podman Desktop (providers, their state being listed)
"Frontend State" of Podman Desktop

Try to do user-driven some operations on the backend like "list containers" and see if it works

Display the status of the current http connection

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Allows to see the logs, try to ping/test container engines

### Screenshot/screencast of this PR

https://github.com/containers/podman-desktop/assets/436777/e0a5deb0-44ad-4eea-9b24-134754fede80

### What issues does this PR fix or reference?

fixes https://github.com/containers/podman-desktop/issues/2858

### How to test this PR?

Click on the bottom right of the status bar on the 💡 icon

### CODE REVIEW

File: packages/renderer/src/lib/diagnose/DiagnoseContainerEngine.svelte [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+<div class="flex flex-col bg-charcoal-800 border-zinc-600 m-4 p-4 text-sm">
+  <h6 aria-label="name">Name: {containerEngineRunning.name}</h6>
+  <div class="mx-4">
+    <h6 aria-label="status">Status: {containerEngineRunning.status}</h6>
+    <h6 aria-label="socket path">Socket path: {containerEngineRunning.endpoint.socketPath}</h6>
+  </div>
+  <div>Actions:</div>
+  <div class="mx-4">
+    <DiagnoseContainerEnginePing providerContainerEngine="{containerEngineRunning}" />
```
Comment (jeffmaury) [inline packages/renderer/src/lib/diagnose/DiagnoseContainerEngine.svelte]: Would be good to vertically align the responses for these 2 elements

File: packages/renderer/src/lib/diagnose/DiagnosePage.svelte [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<script>
+import NavPage from '../ui/NavPage.svelte';
+import DiagnoseDevToolsConsoleLogs from './DiagnoseDevToolsConsoleLogs.svelte';
+import DiagnosePageProviders from './DiagnosePageProviders.svelte';
+</script>
+
+<NavPage title="Self-diagnosis" searchEnabled="{false}">
```
Comment (jeffmaury) [inline packages/renderer/src/lib/diagnose/DiagnosePage.svelte]:
```suggestion
<NavPage title="Self diagnostic" searchEnabled="{false}">
```

**Review by jeffmaury — APPROVED** [review]
Few comments but LGTM

File: packages/renderer/src/lib/diagnose/DiagnoseContainerEngine.svelte (hunk 1 above)
Comment (benoitf) [reply packages/renderer/src/lib/diagnose/DiagnoseContainerEngine.svelte]: image

Comment (benoitf) [reply packages/renderer/src/lib/diagnose/DiagnoseContainerEngine.svelte]: horizontally aligned

File: packages/renderer/src/lib/diagnose/DiagnosePage.svelte (hunk 2 above)
Comment (benoitf) [reply packages/renderer/src/lib/diagnose/DiagnosePage.svelte]: I've renamed the whole content from 'Diagnose' to 'Troubleshooting'

**Pull request merged**